src/vm/stackwalk.cpp

```diff
@@ -1778,7 +1778,7 @@ StackWalkAction StackFrameIterator::Filter(void)
                                 pCurrTracker = pCurrTracker->GetPreviousExceptionTracker();
                             }
 
-                            if (pCurrTracker != NULL)
+                            if ((pCurrTracker != NULL) && !pCurrTracker->IsEHClauseFilter())
```

Comment:
I have tried to run the test on amd64 with GC stress mode C and this "if" was never entered. Could you please verify that the issue really happened in the `JIT/Methodical/eh/basics/throwinfilter_d/throwinfilter_d.exe` test?